libbeat/processors/conditions/conditions.go

```diff
@@ -0,0 +1,109 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package conditions
+
+import (
+	"errors"
+
+	"github.com/elastic/beats/libbeat/common/match"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+// ConditionConfig represents a configuration for a condition, as you would find it in the config files.
+type ConditionConfig struct {
+	Equals    *ConditionFields  `config:"equals"`
+	Contains  *ConditionFields  `config:"contains"`
+	Regexp    *ConditionFields  `config:"regexp"`
+	Range     *ConditionFields  `config:"range"`
+	HasFields []string          `config:"has_fields"`
+	OR        []ConditionConfig `config:"or"`
+	AND       []ConditionConfig `config:"and"`
+	NOT       *ConditionConfig  `config:"not"`
+}
+
+// Condition is the interface for all defined conditions
+type Condition interface {
+	Execute(event ValuesMap) bool
```

Comment:
No strong preference on my end but I'm curious why you renamed it from Check to Execute?

Reply:
It's not a super-strong preference on my side either, but I have a slight preference for execute. 

Check to me sounds like something that checks internal state. Execute sounds better IMHO in the context of an interpreter for an IR/AST.

That said, neither one is right, glad to switch back to Check if you or @urso have a strong preference there, it's a very minor thing.

Reply:
I'd rather use check or something more descriptive. In Go anything implementing `Execute` can be a conditional, as the type checker checks for structural compatibility of values and interface. To me the name `Execute` implies an implementation can do "much more".

Reply:
Cool, I'll switch back to `Check`